metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/external/impl/EntityClientResolver.java

```diff
@@ -0,0 +1,174 @@
+package eu.europeana.enrichment.api.external.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import eu.europeana.enrichment.api.exceptions.UnknownException;
+import eu.europeana.enrichment.api.external.model.*;
+import eu.europeana.enrichment.api.internal.EntityResolver;
+import eu.europeana.enrichment.api.internal.ReferenceTerm;
+import eu.europeana.enrichment.api.internal.SearchTerm;
+import eu.europeana.enrichment.utils.EntityResolverUtils;
+import eu.europeana.entity.client.utils.EntityApiConstants;
+import eu.europeana.entity.client.web.EntityClientApi;
+import eu.europeana.entity.client.web.EntityClientApiImpl;
+
+import eu.europeana.entitymanagement.definitions.model.Entity;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * An entity resolver that works by accessing a service via Entity Client API and obtains entities from
+ * Entity Management API
+ *
+ * @author Srishti.singh@europeana.eu
+ */
+public class EntityClientResolver implements EntityResolver {
+
+    private EntityClientApi entityClientApi;
+    private final int batchSize;
+
+
+    public EntityClientResolver(int batchSize) {
+        this.batchSize = batchSize;
+        entityClientApi = new EntityClientApiImpl();
+    }
+
+    @Override
+    public <T extends SearchTerm> Map<T, List<EnrichmentBase>> resolveByText(Set<T> searchTerms) {
+        final Function<T, EnrichmentQuery> inputFunction = EntityResolverUtils.createEnrichmentQueryForTextSearch();
```

Comment:
How about the 3 and 2 code language check.
In contrary to the fetching parent entities, I don't see the language mapping implemented anyway in this code.

Reply:
I haven't added the ISO language check here. But the languages are validated in entity-api. There is no harm in adding a language validation here, but it will be redundant then.

Reply:
So when you say validated does that mean that either a 2 letter or 3 letter code will produce a result?
If the answer is yes then indeed it would be redundant to add a check here.

Reply:
The check is to validate ISO language ([here](https://github.com/europeana/entity-api-v2/blob/e6c3df4ed92cb457f574de54750c38dc624114a9/entity-web/src/main/java/eu/europeana/entity/web/controller/BaseRest.java#L236)) . I think that validates the two or three-letter code already.

Reply:
I don't think it does the same way metis-enrichment handles it.

Metis does the following:
- If the provided language is a 3 letter code it then converts it to a 2 letter code
- If the provided language is a 2 letter code it checks if it a valid 2 letter code otherwise uses null(which means do not check for a specific language)
- On any other case use null. A null language provided is okay as well.

While the link you mentioned just validates if the language is not null, it does not contain a comma, if it is an exact match to 'all' or it's a valid 2 letter code. If that fails it throws an exception.

So clearly the logic is different.
One of the potential issues here is that 3 letter codes will now fail.
Is this what we want?

Reply:
No, this check should be added: If the provided language is a 3 letter code it then converts it to a 2 letter code. For invalid language codes, send null (no specific language). 

null language is accepted in entity  API.

Reply:
Ok should I implement this here, or do you want this on the Entity API application?

Reply:
After discussion:
The language conversion will be implemented in the ClientEntityResolver in the metis-enrichment libraries.